service/src/main/java/com/myhome/controllers/UserController.java

```diff
@@ -89,6 +97,21 @@
   }
 
   @Override
+  public ResponseEntity<Void> usersPasswordPost(@NotNull @Valid String action, @Valid @RequestBody ForgotPasswordRequest forgotPasswordRequest) {
+    boolean result = false;
+    PasswordActionType parsedAction = PasswordActionType.valueOf(action);
+    if (parsedAction == PasswordActionType.FORGOT) {
+      result = true;
+      userService.requestResetPassword(forgotPasswordRequest);
+    } else if (parsedAction == PasswordActionType.RESET) {
+      result = userService.resetPassword(forgotPasswordRequest);
+    }
+    if (result) {
+      return ResponseEntity.ok().build();
+    } else {
+      return ResponseEntity.badRequest().build();
```

Comment:
In your case badrequest is thrown only when some properties are missing but the Valid annotation checks this.
I would omit that.

Reply:
in case `PasswordActionType.FORGOT` we need only email (`token` and `newPassword` can be empty) and in case `PasswordActionType.RESET` all fiels are requared so I must check null